engine/src/main/java/nl/inl/blacklab/search/results/ResultsStatsDelegate.java

```diff
@@ -0,0 +1,133 @@
+package nl.inl.blacklab.search.results;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import nl.inl.blacklab.exceptions.InterruptedSearch;
+
+/**
+ * Returns 0 until it gets the real stats.
+ *
+ * Useful for peeking at count.
+ */
+public class ResultsStatsDelegate extends ResultsStats {
+
+    /** How often should we poll for the real stats to be available in realStats()? */
+    private static final long STATS_POLL_TIME_MS = 50;
+
+    /** Our cache entry */
+    private final Future<ResultsStats> future;
+
+    /** The actual stats to monitor, as soon as they're available. Null otherwise. */
+    private ResultsStats realStats;
+
+    /** Used to let us wait until the stats are available. */
+    private CountDownLatch realStatsAvailable;
```

Comment:
I think this can also be final here, safer and easier to reasons about in heavy multithreaded applications like blacklab

Reply:
I agree, will do that.

(I tend to forget about final while coding. I used to use Eclipse, which can show up-to-date suggestions in a big list for the entire project, but I've switched to IntelliJ IDEA which seemingly can't, only per-file or by explicitly requesting a Code Inspection, neither of which work for me. There's a lot I like about IntelliJ IDEA, but this is the one thing that is seriously making me consider going back to Eclipse)